proxy/stratum.go

```diff
@@ -51,7 +51,11 @@ func (s *ProxyServer) ListenTCP() {
 			continue
 		}
 		n += 1
-		cs := &Session{conn: conn, ip: ip}
+		cs := &Session{
+			conn:       conn,
+			ip:         ip,
+			Extranonce: fmt.Sprintf("%x", s.rng.Intn(0xffff)),
```

Comment:
whats the reason for this magic initializer?

Reply:
This is the extranonce assigned to this worker. Every time a worker connects a new session is created with a different extranonce.